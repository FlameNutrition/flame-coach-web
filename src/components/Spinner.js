import PropTypes from 'prop-types';
import Page from './Page';
import Container from '@material-ui/core/Container';
import React from 'react';
import makeStyles from '@material-ui/styles/makeStyles';
import Loading from './Loading';
import Warning from './Warning';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.dark,
    minHeight: '100%',
    paddingBottom: theme.spacing(3),
    paddingTop: theme.spacing(3),
    height: '100%'
  },
  container: {
    height: '100%'
  }
}));

const Spinner = ({
  isError,
  isLoading,
  pageTitle,
  children,
}) => {
  const classes = useStyles();

  let elem = <Loading size={100}/>;

  if (!isLoading) {
    if (!isError) {
      elem = children;
    } else {
      elem = <Warning message={process.env.NEXT_PUBLIC_MSG_SERVER_ERROR}/>;
    }
  }

  return (
    <Page
      className={classes.root}
      title={pageTitle}
    >
      <Container
        className={classes.container}
        maxWidth={false}
      >
        {elem}
      </Container>
    </Page>
  );
};

Spinner.propTypes = {
  children: PropTypes.node.isRequired,
  pageTitle: PropTypes.string.isRequired,
  isLoading: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired
};

export default Spinner;
