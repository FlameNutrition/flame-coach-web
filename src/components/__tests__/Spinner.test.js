import Spinner from '../Spinner';
import React from 'react';
import { render, screen } from '../../testing/test-utils';

describe('<Spinner />', () => {

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('Show server error message when isError true', () => {

    render(
      <Spinner
        isError={true}
        isLoading={false}
        pageTitle="Hello Title">
        <div>Hello</div>
      </Spinner>
    );

    expect(screen.queryByText('Oops! Something went wrong! Help us improve your experience by reporting the problem.'))
      .not
      .toBeNull();

  });

  it('Show loading component when isLoading true', () => {
    const { container } = render(
      <Spinner
        isError={false}
        isLoading={true}
        pageTitle="Hello Title">
        <div>Hello</div>
      </Spinner>
    );

    expect(container)
      .toMatchSnapshot();
    expect(screen.queryByText('Oops! Something went wrong! Help us improve your experience by reporting the problem.'))
      .toBeNull();
  });

  it('Show children component when isError and isLoading false', () => {
    render(
      <Spinner
        isError={false}
        isLoading={false}
        pageTitle="Hello Title">
        <div>Hey! Is working...</div>
      </Spinner>
    );

    expect(screen.queryByText('Hey! Is working...'))
      .not
      .toBeNull();
  });

});
