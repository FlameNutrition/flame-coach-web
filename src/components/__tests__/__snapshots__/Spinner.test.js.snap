// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> Show loading component when isLoading true 1`] = `
<div>
  <div
    class="makeStyles-root-12"
  >
    <div
      class="MuiContainer-root makeStyles-container-13"
    >
      <div
        class="MuiBox-root MuiBox-root-14"
      >
        <div
          class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
          role="progressbar"
          style="width: 100px; height: 100px;"
        >
          <svg
            class="MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
              cx="44"
              cy="44"
              fill="none"
              r="20.2"
              stroke-width="3.6"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</div>
`;
